miner: add mine/import tests for non-imaginary configs

meowsbits opened the pull request:
These tests include and extend the imaginary All_ configs
to proving mine/import pattern for configurations that
are actually reflective of real-world configurations.

tzdybal commented on params/confp/generic/generic.go:
```diff
@@ -46,6 +47,9 @@ func (c GenericCC) DAOSupport() bool {
 	if gc, ok := c.ChainConfigurator.(*goethereum.ChainConfig); ok {
 		return gc.DAOForkSupport
 	}
+	if omg, ok := c.ChainConfigurator.(*oldmultigeth.ChainConfig); ok {
```
`omg` is a great name for variable :joy:

tzdybal commented:
My commit make `ChainConfigurator` selection deterministic.

meowsbits committed:
miner: add mine/import tests for non-imaginary configs

These tests include and extend the imaginary All_ configs
to proving mine/import pattern for configurations that
are actually reflective of real-world configurations.

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner: (lint) goimports

Signed-off-by: meows <b5c6@protonmail.com>
